contracts/staking/PoSValidatorManager.sol

```diff
@@ -130,32 +136,83 @@ abstract contract PoSValidatorManager is
         $._rewardCalculator = rewardCalculator;
     }
 
+    function claimDelegationFees(bytes32 validationID) external {
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+
+        Validator memory validator = getValidator(validationID);
+
+        require(
+            validator.status == ValidatorStatus.Completed,
+            "PoSValidatorManager: validation period not completed"
+        );
```

Comment:
Lets keep this requirement for now, but worth calling out it's not technically needed. It would be safe for validators to claim their delegation fees prior to the end of their validation period (it may have tax-related implications for people various jurisdictions though).

Reply:
Yeah, we talked about this on a call yesterday. If we want validators to be able to collect their fees before their validation period ends, we'd have to create a separate map for fees and rewards, because putting them in the same map would allow the validator to collect their rewards in the `PendingRemoved` state.

So we went for this simpler but more restrictive option for now.